lti_consumer/lti_consumer.py

```diff
@@ -860,3 +860,15 @@ def _get_modal_position_offset(self, viewport_percentage):
             float: The css position offset to apply to the modal window
         """
         return (100 - viewport_percentage) / 2
+
+    def get_outcome_service_url(self, service_name="grade_handler"):
+        """
+        This function is called by get_course_lti_endpoints when using LTI result service to
+        discover the LTI result endpoints. It is similar to outcome_service_url property but
+        with a mapping for backward compatibility.
+        """
+        mapping = {
+            'grade_handler': 'outcome_service_handler',
+            'lti_2_0_result_rest_handler': 'result_service_handler'
+        }
+        return self.runtime.handler_url(self, mapping[service_name], thirdparty=True).rstrip('/?')
```

Comment:
Not sure if I understand correctly. The get_outcome_service_url function was missing and it was called from https://github.com/edx/edx-platform/blob/master/lms/djangoapps/courseware/views/views.py#L1075. Sorry, not sure if I understand your comment. The get_outcome_service_url was missing from lti_consumer (was throwing exception when the endpoint was called). It is called from here: https://github.com/edx/edx-platform/blob/master/lms/djangoapps/courseware/views/views.py#L1075. The service_name parameter was specific for LTI xmodule functions (grade_handler and lti_2_0_result_rest_handler). During the refactor to xblock, those function names changed. So that's why we need a mapping.